### ISSUE TITLE
breadcrumbs_trail misbehaves if item identifiers have more than one "."

### ISSUE THREAD

**Issue by iay** [nanoc/nanoc#1278]
Items with more than one "." in their identifier (e.g., "1.6.md") can cause `breadcrumbs_trail` to reference the wrong item.

### Steps to reproduce

Make items like this:

```
/trp/1.5.md
/trp/1.5/foo.md
/trp/1.6.md
/trp/1.6/foo.md
```

The breadcrumbs for one or other of the `foo.md` items will refer to the wrong parent.

### Expected behavior

I'd expect the last element of the breadcrumb trail for an item to always be that item's parent.

### Actual behavior

Sometimes the last element of the breadcrumb trail for an item can be an item with a similar name.

### Details

It looks like the `breadcrumbs_trail` method for full identifiers is stripping extensions for identifier components other than the last: [here](https://github.com/nanoc/nanoc/blob/b322a41d97e6f718d81a8f96aa1a1381620cc92c/nanoc/lib/nanoc/helpers/breadcrumbs.rb#L21)

```ruby
@items[Nanoc::Identifier.new(pr).without_ext + '.*']
```

For example, for `/trp/1.6/software.md`:

* `components` becomes `["", "/trp", "/trp/1.6", "/trp/1.6/software.md"]`

* The identifiers used to construct the prefixes array are: `["/index.*", "/trp.*", "/trp/1.*", "/trp/1.6/software.*"]`

Note that `"/trp/1.*"` can match either `/trp/1.5.md` or `/trp/1.6.md`.

It seems to me as if stripping the extension isn't required here, as long as you _don't_ add the `.*` on to the end of the _last_ identifier in the list: that one will already be present as an explicit extension if we're working with full identifiers.

I could be missing something, though, I am really not a native Ruby programmer (yet).

**Comment by iay** [comment]
Changing this:

```ruby
        prefixes
          .reject { |pr| pr =~ /^\/index\./ }
          .map do |pr|
            if pr == ''
              @items['/index.*']
            else
              @items[Nanoc::Identifier.new(pr).without_ext + '.*']
            end
          end
```

to this:

```ruby
        prefixes.push(Nanoc::Identifier.new(prefixes.pop).without_ext)
        prefixes
          .reject { |pr| pr =~ /^\/index\./ }
          .map do |pr|
            if pr == ''
              @items['/index.*']
            else
              @items[Nanoc::Identifier.new(pr) + '.*']
            end
          end
      end
```

... appears to fix the problem in my case, by only stripping the extension from the last prefix (representing the current item itself). I doubt that it's idiomatic Ruby, and I can't find any unit tests to run to see if it would break something else. But it's a start. I can set up a PR if you want one.

**Comment by iay** [comment]
Alas, that quick fix does indeed run into an additional problem with the `/index.md` item, as it is no longer matched in the `.reject` clause, so you end up with two references to itself from the root item.

This variant appears to address that:

```ruby
        prefixes.push(Nanoc::Identifier.new(prefixes.pop).without_ext)
        prefixes
          .reject { |pr| pr =~ /^\/index$/ }
          .map do |pr|
            if pr == ''
              @items['/index.*']
            else
              @items[Nanoc::Identifier.new(pr) + '.*']
            end
          end
```

**Comment by ddfreyne** [comment]
I believe that the purpose of `#without_ext` was to support items with multiple extensions, like these:

```
/foo/stuff.md
/foo.md.erb
```

**Comment by ddfreyne** [comment]
There is a fix in #1279 with some fancy functional programming ✨
